manifests/init.pp

```diff
@@ -63,4 +63,23 @@
   $syslog_protocol    = 'udp',
   $syslog_port        = 514,
   $syslog_facility    = 'local7',
-) { }
+) {
+  
+  # The hackery is strong.  Not sure where else to hack this in.
+  if $::osfamily == "RedHat" {
+    file{'/etc/facter':
+      ensure  => 'directory',
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0444'
+    }
+
+    file{'/etc/facter/facts.d':
+      ensure  => 'directory',
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0444',
+      require => File['/etc/facter']
+    }
+  }
+}
```

Comment:
I'm not sure how to fix this any better.  We could tweak the location of the boostrap lock files but I dunno.  This comes about because installing puppet/facter on RHEL doesn't generate these directories.